### PR TITLE
Feature/session token

### LINKED ISSUE #18: Implement session token generation
- [x] Generate session token
- [x] Save session token with expiration date: 1 week
- [x] Create issue to implement session token validation logic

### PR DESCRIPTION
**Pull request by rarques**
Closes #18

### CODE REVIEW

**Review by mduenast — APPROVED** [review]

File: src/main/java/com/garannodou/role/infrastructure/authentication/GoogleAuthenticationFacade.java [hunk 1]
```diff
@@ -5,17 +5,28 @@
...
+
 @Service
-public class GoogleAuthenticationService {
-    private UserService userService;
+public class GoogleAuthenticationFacade {
+
+    private final UserService userService;
+    private final JWTGenerator jwtGenerator;
+    private SecretKeyRepository secretKeyRepository;
+    private final String url;
```
Comment (rarques) [inline src/main/java/com/garannodou/role/infrastructure/authentication/GoogleAuthenticationFacade.java]: Rename to meaningful name

File: src/main/java/com/garannodou/role/infrastructure/authentication/GoogleAuthenticationFacade.java [hunk 2]
```diff
@@ -5,17 +5,28 @@
...
+import static com.garannodou.role.infrastructure.authentication.JWTGenerator.TokenGenerationParams;
+
 @Service
-public class GoogleAuthenticationService {
-    private UserService userService;
+public class GoogleAuthenticationFacade {
+
+    private final UserService userService;
+    private final JWTGenerator jwtGenerator;
+    private SecretKeyRepository secretKeyRepository;
```
Comment (rarques) [inline src/main/java/com/garannodou/role/infrastructure/authentication/GoogleAuthenticationFacade.java]: Move to JWTGenerator

File: src/test/java/com/garannodou/role/infrastructure/authentication/JWTGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        // Then
+        Claims claims = decodeJWT(key, token);
+        assertThat(claims.getIssuer()).isEqualTo("localhost");
+        assertThat(UUID.fromString(claims.getSubject())).isEqualTo(USER_ID);
+        assertThat(claims.get("scope")).isEqualTo("user");
+    }
+
+    @Test
+    public void shouldFailIfTokenIsSignedWithDifferentKey() {
```
Comment (rarques) [inline src/test/java/com/garannodou/role/infrastructure/authentication/JWTGeneratorTest.java]: shouldFailWhenDecodingIfTokenIsSignedWithDifferentKey

File: src/test/java/com/garannodou/role/infrastructure/authentication/SecretKeyRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import javax.crypto.SecretKey;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SecretKeyRepositoryTest {
+
+    @Test
+    public void test() throws Exception {
```
Comment (rarques) [inline src/test/java/com/garannodou/role/infrastructure/authentication/SecretKeyRepositoryTest.java]: Change name

**Review by mduenast — APPROVED** [review]

**Pull request merged**